src/memory_target_ops.h

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifndef UMF_MEMORY_TARGET_OPS_H
+#define UMF_MEMORY_TARGET_OPS_H 1
+
+#include <stdbool.h>
+
+#include <umf/base.h>
+#include <umf/memspace.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct umf_memory_target_t *umf_memory_target_handle_t;
+
+struct umf_memory_target_ops_t {
+    /// Version of the ops structure.
+    /// Should be initialized using UMF_VERSION_CURRENT
+    uint32_t version;
+
+    enum umf_result_t (*initialize)(void *params, void **mem_target);
+    void (*finalize)(void *mem_target);
+
+    enum umf_result_t (*pool_create_from_memspace)(
+        umf_memspace_handle_t memspace, void **mem_targets, size_t num_targets,
```

Comment:
If memspace is a set of memory targets why do we need to pass `mem_targets` array and its size (`num_targets`) as additional parameters?

Reply:
This is because `mem_targets` are pointers to private memory target data (to the structure that was initialized by memory_target_ops::initialize) whereas memspace->nodes points to the internal `struct umf_memory_target_t`. See here how they are translated: https://github.com/oneapi-src/unified-memory-framework/blob/ad096207c6ff8a9e7fa098c39682e1de7e56f4a7/src/memspace.c#L58

This is not strictly necessary for now but it will be necessary when we decide to expose memory_target_ops as public interface in the future (we don't want to leak `struct umf_memory_target_t` definition).